### ISSUE TITLE
Printing from PDF

### ISSUE THREAD

**Issue by rangathedon** [python-escpos/python-escpos#376]
Can this library be used to print a pdf file on disk directly? could not find any example on the same; please help

**Comment by patkan** [comment]
No, this is not supported.
You could try to use a windows printer and redirect this to PDF, be however aware, that this would print the raw escpos. There is no way (inside this software) to "render" escpos. I am closing this, but feel free to give notice if anything else can be done.

**Comment by rangathedon** [comment]
My question was not to generate a PDF; my question was if I could give a PDF file name and print it like its allowed for images!

**Comment by patkan** [comment]
OK, it sounded like you wanted to print to a PDF file onto disk.
For reading and the printing it, you would have to load it into PILlow. So anything that can be loaded by this class would be good: https://github.com/python-escpos/python-escpos/blob/51d1299285c8527b7863bf04fb34ce65ccf5ef30/src/escpos/image.py#L20
I don't know, if it is possible to raster a PDF with PIL, but if you have an exteral way of rendering it and then storing a PIL-object, you could directly pass it to that function.

**Comment by patkan** [comment]
So maybe this could be a starter that you can try: https://pypi.org/project/pdf2image/

**Comment by rangathedon** [comment]
I tried the suggestion, it seems to work though a bit slow in dumping pdf to disk as image;

Is there any way to set margins for the printer to 0? the image content already has proper margins but its getting cut off when it prints the image

**Comment by patkan** [comment]
If it works you could put together a short example that we could add to the manual. I am not sure how this library works, but as far as I see it it should also be possible to get directly a PIL object from it. In this case you would not need to save to disk, but could call the image function with this PIL object.

If you already print in the vincinity of the edge of the paper, there is probably no way to increase the marging further. The only hacky solution that I could think of at the moment would be cutting the marging from the image (it is a PIL object after all, so image manipulation should be rather easy at this point).

**Comment by rangathedon** [comment]
Nope, its wasting more margin than needed, thats the reason I wanted to know if I could set print margins in this LIB; more than 5mm margin is going on both sides; I believe this might be hardcoded somewhere; ideally this should be a parameter to be set somewhere isnt it?

On the example, there is nothing special by what was strange is that when I tried to print PIL image directly by passing it to escpos lib, it printed raw image hex data as text so I ended up writing the PIL image to disk as a JPEG and then printed it using escpos lib;

**Comment by hasansezertasan** [comment]
Is this feature in-scope of `python-escpos`?

Maybe a third-party library could be a better solution for this.
